packages/components/src/components/contract-card/contract-card-items/toggle-card-dialog.jsx

```diff
@@ -12,43 +12,69 @@ import Popover from '../../popover';
 import Div100vhContainer from '../../div100vh-container';
 import './sass/contract-card-dialog.scss';
 
-class ToggleCardDialog extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            is_visible: false,
-            top: 0,
-            left: 0,
-            is_do_not_show_selected: !this.props.should_show_cancellation_warning,
-        };
-        this.toggle_ref = React.createRef();
-        this.dialog_ref = React.createRef();
-        this.contract = props.getContractById(props.contract_id);
-        this.ContractUpdateFormWrapper = props.connectWithContractUpdate?.(ContractUpdateForm) || ContractUpdateForm;
-    }
-
-    handleClick = e => {
+let ContractUpdateFormWrapper;
+
+const ToggleCardDialog = ({
+    addToast,
+    contract_id,
+    connectWithContractUpdate,
+    current_focus,
+    getCardLabels,
+    getContractById,
+    is_valid_to_cancel,
+    removeToast,
+    setCurrentFocus,
+    should_show_cancellation_warning,
+    status,
+    toggleCancellationWarning,
+}) => {
+    const [is_visible, setIsVisible] = React.useState(false);
+    const [top, setTop] = React.useState(0);
+    const [left, setLeft] = React.useState(0);
+    const [is_do_not_show_selected, setIsDoNotShowSelected] = React.useState(!should_show_cancellation_warning);
+
+    const toggle_ref = React.useRef();
+    const dialog_ref = React.useRef();
+    const contract = getContractById(contract_id);
+    ContractUpdateFormWrapper =
+        ContractUpdateFormWrapper || connectWithContractUpdate?.(ContractUpdateForm) || ContractUpdateForm;
+
+    React.useEffect(() => {
+        if (is_visible && toggle_ref && toggle_ref.current && dialog_ref && dialog_ref.current) {
+            const icon_bound = toggle_ref.current.getBoundingClientRect();
+            const portal_ref = dialog_ref.current;
+            const target_bound = portal_ref?.current?.getBoundingClientRect();
+            const body_bound = document.body.getBoundingClientRect();
+
+            let { top: icon_bound_top } = icon_bound;
+            const { right } = icon_bound;
+
+            if (icon_bound_top + target_bound?.height > body_bound.height) {
+                icon_bound_top -= target_bound?.height - icon_bound.height;
+            }
+
+            setTop(icon_bound_top);
+            setLeft(right - 16);
+        }
+    }, [is_visible]);
```

Comment:
Have you checked eslint warnings here for `exhaustive deps`?

Reply:
Nothing showed up in test.